drivers/adc/adc_ambiq.c

```diff
@@ -0,0 +1,388 @@
+{
+	struct adc_ambiq_data *data = CONTAINER_OF(ctx, struct adc_ambiq_data, ctx);
+
+	if (repeat_sampling) {
+		data->buffer = data->repeat_buffer;
+	}
+}
+
+static void adc_context_start_sampling(struct adc_context *ctx)
+{
+	struct adc_ambiq_data *data = CONTAINER_OF(ctx, struct adc_ambiq_data, ctx);
+
+	data->repeat_buffer = data->buffer;
+	/*Trigger the ADC*/
+	am_hal_adc_sw_trigger(data->adcHandle);
+}
+
+static int adc_ambiq_init(const struct device *dev)
+{
+	struct adc_ambiq_data *data = dev->data;
+	const struct adc_ambiq_config *cfg = dev->config;
+	am_hal_adc_config_t ADCConfig;
+
+	int ret;
+
+	/* Initialize the ADC and get the handle*/
+	if (AM_HAL_STATUS_SUCCESS !=
+	    am_hal_adc_initialize((cfg->base - REG_ADC_BASEADDR) / (cfg->size * 4),
+				  &data->adcHandle)) {
+		ret = -ENODEV;
+		LOG_ERR("Faile to initialize ADC, code:%d", ret);
+		return ret;
+	}
+
+	/* power on ADC*/
+	ret = cfg->pwr_func();
+
+	/* Set up the ADC configuration parameters. These settings are reasonable
+	 *  for accurate measurements at a low sample rate.
+	 */
+	ADCConfig.eClock = AM_HAL_ADC_CLKSEL_HFRC;
+	ADCConfig.ePolarity = AM_HAL_ADC_TRIGPOL_RISING;
+	ADCConfig.eTrigger = AM_HAL_ADC_TRIGSEL_SOFTWARE;
+	ADCConfig.eReference = AM_HAL_ADC_REFSEL_INT_1P5;
+	ADCConfig.eClockMode = AM_HAL_ADC_CLKMODE_LOW_POWER;
+	ADCConfig.ePowerMode = AM_HAL_ADC_LPMODE0;
+	ADCConfig.eRepeat = AM_HAL_ADC_SINGLE_SCAN;
+	if (AM_HAL_STATUS_SUCCESS != am_hal_adc_configure(data->adcHandle, &ADCConfig)) {
+		ret = -ENODEV;
+		LOG_ERR("Configuring ADC failed, code:%d", ret);
+		return ret;
+	}
+
+	ret = pinctrl_apply_state(cfg->pin_cfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Enable the ADC interrupts in the ADC. */
+	cfg->irq_config_func();
+	am_hal_adc_interrupt_enable(data->adcHandle, AM_HAL_ADC_INT_CNVCMP);
+
+	adc_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int adc_ambiq_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	struct adc_ambiq_data *data = dev->data;
+	uint32_t ret;
+	am_hal_sysctrl_power_state_e status;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		status = AM_HAL_SYSCTRL_WAKE;
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		status = AM_HAL_SYSCTRL_DEEPSLEEP;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	ret = am_hal_adc_power_control(data->adcHandle, status, true);
+
+	if (ret != AM_HAL_STATUS_SUCCESS) {
+		return -EPERM;
+	} else {
+		return 0;
+	}
+}
+#endif /* CONFIG_PM_DEVICE */
+
+#define ADC_AMBIQ_DRIVER_API(n)                                                                    \
+	static const struct adc_driver_api adc_ambiq_driver_api_##n = {                            \
+		.channel_setup = adc_ambiq_channel_setup,                                          \
+		.read = adc_ambiq_read,                                                            \
+		.ref_internal = DT_INST_PROP(n, vref_mv),                                          \
```

Comment:
@decsny can you tell me what adc_driver_api.ref_internal is used for?
I do not see a use in Zephyr?